lib/model/message_list.dart

```diff
@@ -420,6 +420,35 @@ class MessageListView with ChangeNotifier, _MessageSequence {
     notifyListeners();
   }
 
+  void maybeUpdateMessageFlags(UpdateMessageFlagsEvent event) {
+    final isAdd = switch (event) {
+      UpdateMessageFlagsAddEvent()    => true,
+      UpdateMessageFlagsRemoveEvent() => false,
+    };
+
+    bool didUpdateAny = false;
+    if (event.all) {
+      for (final message in messages) {
+        (isAdd ? message.flags.add : message.flags.remove).call(event.flag);
```

Comment:
It looks like `call` is dynamically typed. If for example I replace `remove` with `join` here — a random other method that should take a different type of argument — then there's no error.

Can clean up by expanding things to be just slightly more explicit:
```dart
        isAdd ? message.flags.add(event.flag) : message.flags.remove(event.flag);
```
which also has the virtue of looking more like conventional function calls we have in the rest of our code.